### PR TITLE
Update sphinx

### PR DESCRIPTION
**Pull request by A5rocks**
I was looking at https://github.com/python-trio/trio/issues/2921 and wondering why Sphinx wasn't getting updated. Turns out they dropped support for Python 3.8. However, our RTD build runs in Python 3.11 so we can use that for the docs requirements.

### CODE REVIEW

**Comment by A5rocks** [comment]
```
/home/docs/checkouts/readthedocs.org/user_builds/trio/checkouts/3070/src/trio/abc.py:docstring of trio.abc.Channel:1: WARNING: py:class reference target not found: trio._abc.SendChannel
/home/docs/checkouts/readthedocs.org/user_builds/trio/checkouts/3070/src/trio/abc.py:docstring of trio.abc.Channel:1: WARNING: py:class reference target not found: trio._abc.ReceiveChannel
/home/docs/checkouts/readthedocs.org/user_builds/trio/checkouts/3070/src/trio/__init__.py:docstring of trio.SocketListener:1: WARNING: py:class reference target not found: trio._abc.Listener
/home/docs/checkouts/readthedocs.org/user_builds/trio/checkouts/3070/src/trio/__init__.py:docstring of trio.SSLListener:1: WARNING: py:class reference target not found: trio._abc.Listener
/home/docs/checkouts/readthedocs.org/user_builds/trio/checkouts/3070/src/trio/__init__.py:docstring of trio.SSLListener:1: WARNING: py:class reference target not found: trio._ssl.SSLStream
/home/docs/checkouts/readthedocs.org/user_builds/trio/checkouts/3070/src/trio/__init__.py:docstring of trio.DTLSChannel:1: WARNING: py:class reference target not found: trio._abc.Channel
/home/docs/checkouts/readthedocs.org/user_builds/trio/checkouts/3070/src/trio/lowlevel.py:docstring of trio.lowlevel.Task.iter_await_frames:1: WARNING: py:class reference target not found: types.FrameType
```

**Comment by A5rocks** [comment]
Ok most of those is that Sphinx doesn't seem to like referencing a superclass. That makes no sense!

**Comment by A5rocks** [comment]
OK filed https://github.com/python/cpython/issues/123250, but there's things we can do to work around this.

**Review by CoolCat467 — APPROVED** [review]
Looks good. https://github.com/python/cpython/issues/123250 is quite interesting, I hope there will be a fix eventually. I was wondering while looking at this, any particular reason to use 3.11 over 3.12 now that we are upgrading?

File: docs-requirements.in [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # sphinx 5.3 doesn't work with our _NoValue workaround
 sphinx >= 6.0
 jinja2
-sphinx_rtd_theme
+sphinx_rtd_theme >= 2
```
Comment (jakkdl) [inline docs-requirements.in:5]: could add a short comment explaining the `>=` requirement

**Review by jakkdl — APPROVED** [review]
sad to need ugly workarounds, but seems worth it

File: docs-requirements.in (hunk 1 above)
Comment (A5rocks) [reply docs-requirements.in:5]: Yep, it's just cause how `uv` locks things, can definitely leave a comment.

**Pull request merged**